FIX: premapper bugs with AND and XOR gate, sortTopologically add in umain after premapping. Small fixes in namespaces. ADD: ril xag tests

NikitaRM21 opened the pull request.

NikitaRM21 committed:
add: AIG mapper tests

NikitaRM21 committed:
Merge branch 'mvg-internship:main' into main

NikitaRM21 committed:
ADD: AIG tests, REF: XAG tests

NikitaRM21 committed:
Merge branch 'mvg-internship:main' into main

NikitaRM21 committed:
FIX: premapper bugs with AND and XOR gate, sortTopologically add in umain after premapping. Small fixes in namespaces.

NikitaRM21 committed:
ADD: ril XAG tests added

Commit:
fixing tests (MIG, XMG)
using namespace was deleted in gnet_test

Commit:
Merge branch 'main' of https://github.com/NikitaRM21/utopia-eda

Commit:
fix: xag_ril_test

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,183 @@
+#include <random>
+
+/*
+ * Check basis gate correct mapping
```
Комментарий малопонятен. Что такое "basis gate"? Как проверяется корректность? Лучше написать, что это тестовый набор для (видимо) AIG-premapper'а.

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,183 @@
+ */
+
+TEST(AigPremapperTest, AigNotOneInputTest) {
+  // Create network
```
Здесь и далее этот комментарий лучше удалить

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,183 @@
+TEST(AigPremapperTest, AigNotOneInputTest) {
+  // Create network
+  std::shared_ptr<GNet> net = makeSingleGateNet(GateSymbol::NOT, 1);
+  // Premapping
```
Здесь и далее этот комментарий лучше удалить

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,183 @@
+  // Premapping
+  GateIdMap gmap;
+  std::shared_ptr<GNet> premapped = premap(net, gmap, PreBasis::AIG);
+  // Check equivalence
```
Здесь и далее этот комментарий лучше удалить

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,183 @@
+}
+
+/*
+ * Check out of basis gate correct mapping
```
Аналогично замечанию выше.

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,183 @@
+}
+
+/*
+ * Check basis gate inverted inputs correct mapping
```
Аналогично замечанию выше.

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,183 @@
+}
+
+/*
+ * Check out of basis gate inverted inputs correct mapping
```
Аналогично замечанию выше.

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,183 @@
+  // Check equivalence
+  EXPECT_TRUE(checkEquivalence(net, premapped, gmap));
+}
+
```
Лишняя пустая строчка -- удалить

ssmolov commented on test/gate/premapper/mapper/mapper_test.h:
```diff
@@ -8,26 +8,26 @@
 #include "gate/premapper/xagmapper.h"
-
+#include "gate/premapper/xmgmapper.h"
+#include "gate/debugger/checker.h"
```
Нарушение алфавитного порядка include

ssmolov commented on test/gate/premapper/migmapper/migmapper_test.cpp:
```diff
@@ -6,6 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "gate/premapper/mapper/mapper_test.h"
```
Нарушение алфавитного порядка include

ssmolov commented on test/gate/premapper/xagmapper/xag_ril_test.cpp:
```diff
@@ -0,0 +1,160 @@
+#include <memory>
+#include <sstream>
+#include <string>
+
```
Лишняя пустая строка.Вместо неё лучше напишите общий комментарий о том, что проверяют тесты в данном файле.

ssmolov commented on test/gate/premapper/xagmapper/xag_ril_test.cpp:
```diff
@@ -0,0 +1,160 @@
+  return context.equal;
+}
+
+int rtlMain(RtlContext &context) {
```
Здесь вы практически скопипастили rtlMain из main.cpp. Зачем?

Всякий раз, когда хотите что-то скопировать, задумайтесь о том, как этого избежать. Если идей нет, спросите товарищей.

Commit:
ref: tests for premappers (AIG, MIG, XAG ril)

ssmolov commented on test/gate/premapper/xagmapper/xag_ril_test.cpp:
```diff
@@ -0,0 +1,158 @@
+#include <string>
+
+// RIL tests for XAG-mapper
+// Verify equivalence of premapped networks for arithmetic operations '+', '-', '*' described on ril
```
1. ril -> RIL (не коверкайте имена собственные, этого никто не любит)

ssmolov commented on test/gate/premapper/aigmapper/aig_test.cpp:
```diff
@@ -0,0 +1,121 @@
+#include <cassert>
+#include <memory>
+#include <random>
+
```
Попросил написать адекватный заголовочный комментарий вместо прежнего -- в результате, его просто удалили. Зря.

ssmolov commented on test/gate/premapper/xagmapper/xag_ril_test.cpp:
```diff
@@ -0,0 +1,158 @@
+  EXPECT_TRUE(rtlMain(context));
+}
+
+//BDD have no dff trigger gate support!
```
Поставьте хотя бы TODO здесь ("// TODO: BDD have no dff trigger gate support!") -- иначе есть риск, что закомментированный тест никто никогда больше не расккомментирует и не узнает, что не поддерживаются DFF triggers. Кстати, об этой проблеме лучше бы завести issue.

Commit:
fix: migmapper, xmgmapper
ref: xag_ril_test

NikitaRM21 committed:
FIX: bug in mapAnd contraty case, ADD: new xag test with contrary inputs

Commit:
rtl_context.[h/cpp] files were added
src/tool/CMakeLists.txt was added
fix: main.cpp, xag_ril_test.cpp
src/CMakeLists.txt: Tool library was added

Commit:
Merge branch 'main' of https://github.com/NikitaRM21/utopia-eda

Commit:
fix: migmapper.cpp

NikitaRM21 committed:
FIX: OUT gate sharing

Commit:
supported MAJ gate symbol in encoder.[h/cpp], small fixes

Commit:
Merge branch 'main' of https://github.com/NikitaRM21/utopia-eda

NikitaRM21 committed:
ADD: tests for XAG mapper based on ril arithmetic schemes with all types of checkers.

ssmolov commented on src/tool/rtl_context.cpp:
```diff
@@ -0,0 +1,107 @@
+
+namespace eda::tool {
+
+void dump(const GNet &net) {
```
Место dump-функции точно не в rtl_context. Не надо её здесь держать.

ssmolov commented on src/tool/rtl_context.h:
```diff
@@ -0,0 +1,70 @@
+using XmgMapper = eda::gate::premapper::XmgMapper;
+
+namespace eda::tool {
+
```
Добавьте краткий комментарий, описывающий семантику структуры RtlContext.

ssmolov commented on test/gate/premapper/mapper/mapper_test.cpp:
```diff
@@ -0,0 +1,80 @@
+  return net;
+}
+
+// gate(x1,~x2).
```
По аналогии с комментариями выше, поставьте пробел перед "~"

ssmolov commented on test/gate/premapper/xagmapper/xag_test.cpp:
```diff
@@ -23,130 +23,50 @@ using GateIdMap = std::unordered_map<Gate::Id, Gate::Id>;
  * Check basis gate correct mapping
  */
 
+
```
Лишняя пустая строка

Commit:
utils.[h/cpp] function "dump" was added
gnet_test, rtl_context function "dump" was removed
ref: rtl_context, mapper_test, xag_test

Commit:
Merge branch 'main' of https://github.com/NikitaRM21/utopia-eda

NikitaRM21 committed:
Merge branch 'main' into main
